baybe/utils/plotting.py

```diff
@@ -140,3 +140,60 @@ def create_example_plots(
         else:
             warnings.warn("Plots could not be saved.")
     plt.close()
+
+
+def indent(text: str, amount: int = 3, ch: str = " ") -> str:
+    """Indent a given text by a certain amount."""
+    padding = amount * ch
+    return "".join(padding + line for line in text.splitlines(keepends=True))
+
+
+def to_string(
+    header: str,
+    *fields: Any,
+    single_line: bool = False,
+    allowed_header_chars: list[str] = (" "),
+) -> str:
+    """Create a nested string representation.
+
+    Args:
+        header: The header, typically the name of a class.
+        *fields: Fields to be printed with an indentation.
+        single_line: If ``True``, print the representation on a single line.
+            Only applicable when given a single field.
+        allowed_header_chars: non-alphanumeric characters allowed in the header.
+
+    Raises:
+        ValueError: If ``single_line`` is ``True`` but ``fields`` contains more than one
+            element.
+        ValueError: If ``header`` contains non-alphanumeric characters other than
+            allowed via `àllowed_header_chars``.
+
+    Returns:
+        The string representation with indented fields.
+    """
+    # If the header is a single line, we verify that it is alpha numeric (with the
+    # exception of it having potential blanks). This check is only done for single line
+    # headers since header can be a more complex object when using this function to
+    # get the str representation of parent classes.
+    if "\n" not in header:
+        # Check whether header only contains alpha numeric values or spaces
+        header_replace = header
+        for char in allowed_header_chars:
+            header_replace = header_replace.replace(char, "")
+        if not header_replace.replace(" ", "").isalnum():
+            raise ValueError(
+                "Only alphanumeric characters and spaces are allowed in ``header``,"
+                f" but header was {header}"
+            )
+        # Append ":"
+        header = header + ":"
```

Comment:
With the chance that you are gonna kill me now ... 🙈 

I think this part is really getting out of hands, to be honest:
* The logic is both substantially longer AND harder to understand than the part of the part of the function for which is actually designed for
* Types are messed up (you are saying `list[str]` but are not actually passing one as default, nor should it be a list in the first place)
* Your comments are hardcoding things that are no longer true when passing something different than the default
* The fact that the check is only done for single-line is coupling the logic to assumption about the input in other situations
* The entire setting is now really be a "katze beißt sich in den eigenen Schwanz" kind of logic, because the control of the input validation comes from the same place that also *provides* the input to be validates ... so a sort of "die bank macht ihre eigene wirtschaftsprüfung" if you know what I mean.

Therefore, honestly, I would just say: let's drop it. But will you still talk to me afterwards? 😨

Reply:
Will then remove any checks and modifications to header completely.